### ISSUE TITLE
[no-input-rename] False positive when directive apply on specific selector

### ISSUE THREAD

**Issue by noemi-salaun** [angular-eslint/angular-eslint#978]
eslint reports an error when I rename an `@Input` to match the directive name but the directive is applied only on specific element like `img[myDirective]`

```JSON
{
  "rules": {
    "@angular-eslint/no-input-rename": [true]
  }
}
```

```TS
@Directive({
  selector: 'img[appPicture]',
})
export class ImgPictureDirective {
  @Input('appPicture') picture: Picture; // <-- ESLint: Input bindings should not be aliased

  @HostBinding() get src(): string {
    return this.picture.src;
  }
}
```

The error disapears if I change my directive selector to only `[appPicture]`

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/eslint-plugin`    | `13.2.0` |
| `@typescript-eslint/parser`        | `13.2.0` |
| `ESLint`                           | `8.7.0` |

```sh
Angular CLI: 13.2.6
Node: 16.14.2
Package Manager: npm 8.5.0
OS: linux x64

Angular: 13.2.7
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1302.6
@angular-devkit/build-angular   13.2.6
@angular-devkit/core            13.2.6
@angular-devkit/schematics      13.2.6
@angular/cdk                    13.2.6
@angular/cli                    13.2.6
@angular/flex-layout            13.0.0-beta.38
@angular/material               13.2.6
@schematics/angular             13.2.6
ng-packagr                      13.3.0
rxjs                            7.4.0
typescript                      4.5.5
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by JamesHenry** [comment]
Thanks @noemi-salaun, it makes sense, and happy to accept a PR to address this

**Comment by adbaran1** [comment]
@noemi-salaun @JamesHenry #1207 makes an attempt to address this, and is ready for review.
